src/Api/WebSocket/MarketData/BookKeeper.php

```diff
@@ -90,16 +105,16 @@ private function processMessage(\stdClass $payload)
         $this->socketSequence = $payload->socket_sequence;
     }
 
-    public function getWebSocketUrl() : string
+    public function getWebSocketUrl(): string
     {
         $str = \sprintf(
             'wss://%s%s%s?',
-            (new Config())->gemini->api->host,
+            Config::getInstance()->get('gemini')->api->host,
             self::API_PATH,
             $this->pair->getPairSymbol()
         );
-        for ($i = 0, $j = \count($this->params); $i < $j; $i++) {
-            $str .= \array_keys($this->params)[$i] . '=' . \array_values($this->params)[$i] . '&';
+        for ($i = 0, $j = \count(self::PARAMS); $i < $j; $i++) {
```

Comment:
`+1` for instantiating $j rather than invoking `\count()` with every iteration. I do the same ever since BMI.